pkg/subscription/clientid/clientid.go

```diff
@@ -1,154 +1,64 @@
 package clientid
 
 import (
-	"fmt"
-
-	"github.com/Axway/agent-sdk/pkg/cache"
-
 	"github.com/Axway/agent-sdk/pkg/apic"
 	"github.com/Axway/agents-mulesoft/pkg/anypoint"
 	"github.com/Axway/agents-mulesoft/pkg/config"
 	"github.com/Axway/agents-mulesoft/pkg/subscription"
-	"github.com/sirupsen/logrus"
 )
 
-type clientId struct {
-	apc anypoint.Client
-}
+const name = "client-id-enforcement"
 
-const (
-	name = "client-id-enforcement"
-)
+type clientID struct {
+	name   string
+	schema apic.SubscriptionSchema
+}
 
 func init() {
-	subscription.Register(func(apc anypoint.Client) subscription.SubscribeHandler {
-		return &clientId{apc: apc}
+	subscription.Register(func(client anypoint.Client) subscription.Contract {
+		return subscription.NewContractBase(client, NewClientIDContract(client))
 	})
 }
 
-func (c *clientId) IsApplicable(pd config.PolicyDetail) bool {
-	return pd.Policy == apic.Apikey && pd.IsSlaBased == false
+// NewClientIDContract creates a new subscribable contract for the client-id policy
+func NewClientIDContract(client anypoint.Client) subscription.Contract {
+	return subscription.NewContractBase(client, newClientID())
 }
 
-func (c *clientId) Schema() apic.SubscriptionSchema {
+// newClientID Creates a newClientID *clientID
+func newClientID() *clientID {
 	schema := apic.NewSubscriptionSchema(name)
 
 	schema.AddProperty(anypoint.AppName,
 		"string",
 		"Name of the new app",
 		"",
 		true,
-		nil)
+		nil,
+	)
 
 	schema.AddProperty(anypoint.Description,
 		"string",
 		"Description",
 		"",
 		false,
-		nil)
-
-	return schema
-}
-
-func (c *clientId) Name() string {
-	return name
-}
-
-func (c *clientId) Subscribe(log logrus.FieldLogger, subs apic.Subscription) error {
-	clientID, clientSecret, err := c.doSubscribe(log, subs)
+		nil,
+	)
 
-	if err != nil {
-		log.WithError(err).Error("Failed to subscribe")
-		return subs.UpdateState(apic.SubscriptionFailedToSubscribe, err.Error())
+	return &clientID{
+		name:   name,
+		schema: schema,
 	}
-
-	return subs.UpdateStateWithProperties(apic.SubscriptionActive, "", map[string]interface{}{
-		anypoint.ClientIDProp: clientID, anypoint.ClientSecretProp: clientSecret})
 }
 
-func (c *clientId) doSubscribe(log logrus.FieldLogger, subs apic.Subscription) (string, string, error) {
```

Comment:
clientId and slaTier no longer have the subscription funcs built into them since their code was the same. They have been moved to the `ContractBase` struct.